### ISSUE TITLE
switching on boolean should not require `else` prong if both `true` and `false` prongs are present

### ISSUE THREAD

**Issue by tgschultz** [ziglang/zig#1768]
```
pub fn init(out_stream: *Stream) Self
{
    return Self
    {
        .out_stream = switch(is_packed)
        {
            true => BitOutStream(endian, Stream.Error).init(out_stream),
            false => out_stream,
        },
    };
}
```

```
error: else prong required when switching on type 'bool'
```

**Comment by andrewrk** [comment]
workaround: use an if statement

**Comment by tgschultz** [comment]
Alternative: use `else` instead of `false`. The reason I went with a `switch` instead of an `if` here is that the latter is, in my opinion, slightly harder to read:

```
pub fn init(out_stream: *Stream) Self
{
    return Self
    {
        .out_stream = blk:
        {
            if(is_packed) break :blk BitOutStream(endian, Stream.Error).init(out_stream);
            break :blk out_stream;
        },
    };
}
```

**Comment by andrewrk** [comment]
Assuming you're ok with using 105 columns, here's how I'd write it:

```zig
pub fn init(out_stream: *Stream) Self {
    return Self{
        .out_stream = if (is_packed) BitOutStream(endian, Stream.Error).init(out_stream) else out_stream,
    };
}
```

or

```zig
pub fn init(out_stream: *Stream) Self {
    return Self{
        .out_stream = if (is_packed)
            BitOutStream(endian, Stream.Error).init(out_stream)
        else
            out_stream,
    };
}
```

`zig fmt` is a bit hostile to both of those, which I'll make a note to myself to potentially remedy.

Regardless, it makes sense to make switch just work with bool types. Thanks for the report.

**Comment by kristate** [comment]
This was fixed in #1857 -- please close.
